File: solutions/java/resistor-color-trio/1/src/main/java/ResistorColorTrio.java
import java.util.*;

class ResistorColorTrio {
    private Map<String, Integer> colorMap = new LinkedHashMap<String, Integer>() {{
        put("black", 0);
        put("brown", 1);
        put("red", 2);
        put("orange", 3);
        put("yellow", 4);
        put("green", 5);
        put("blue", 6);
        put("violet", 7);
        put("grey", 8);
        put("white", 9);
    }};

    int colorCode(String color) {
        return colorMap.get(color);
    }

    String label(String[] colors) {
        int baseValue = colorCode(colors[0]) * 10 + colorCode(colors[1]);
        long totalValue = baseValue * (long)Math.pow(10, colorCode(colors[2]));
        if (totalValue >= 1000000000) {
            return (totalValue / 1000000000) + " gigaohms";
        } else if (totalValue >= 1000000) {
            return (totalValue / 1000000) + " megaohms";
        } else if (totalValue >= 1000) {
            return (totalValue / 1000) + " kiloohms";
        } else {
            return totalValue + " ohms";
        }
    }
}

File: solutions/python/resistor-color-trio/1/resistor_color_trio.py
def label(colors):
    color_codes = {
        "black": 0,
        "brown": 1,
        "red": 2,
        "orange": 3,
        "yellow": 4,
        "green": 5,
        "blue": 6,
        "violet": 7,
        "grey": 8,
        "white": 9
    }
    first_digit = color_codes[colors[0]]
    second_digit = color_codes[colors[1]]
    multiplier = color_codes[colors[2]]
    value = int(str(first_digit) + str(second_digit)) * (10 ** multiplier)
    
    if value >= 1_000_000_000:
        label_value = value // 1_000_000_000
        label_str = f"{label_value} gigaohms"
    elif value >= 1_000_000:
        label_value = value // 1_000_000
        label_str = f"{label_value} megaohms"
    elif value >= 1_000:
        label_value = value // 1_000
        label_str = f"{label_value} kiloohms"
    else:
        label_str = f"{value} ohms"
    return label_str
